Explorer/Assets/DCL/Passport/Modules/EquippedItems_PassportModuleController.cs

```diff
@@ -0,0 +1,316 @@
+                equippedWearableItem.RarityLabelText.text = rarityName;
+                equippedWearableItem.RarityLabelText.color = rarityColor;
+                equippedWearableItem.RarityBackground2.color = new Color(rarityColor.r, rarityColor.g, rarityColor.b, equippedWearableItem.RarityBackground2.color.a);
+                equippedWearableItem.FlapBackground.color = rarityColor;
+                equippedWearableItem.CategoryImage.sprite = categoryIcons.GetTypeImage("emote");
+                string marketPlaceLink = GetMarketplaceLink(emote.GetUrn());
+                equippedWearableItem.BuyButton.interactable = emote.IsCollectible() && rarityName != "base" && marketPlaceLink != string.Empty;
+                equippedWearableItem.BuyButton.onClick.AddListener(() => webBrowser.OpenUrl(marketPlaceLink));
+                LayoutRebuilder.ForceRebuildLayoutImmediate(equippedWearableItem.RarityLabelContainer);
+                WaitForThumbnailAsync(emote, equippedWearableItem, cts.Token).Forget();
+                instantiatedEquippedItems.Add(equippedWearableItem);
+                elementsAddedInTheGird++;
+            }
+
+            int missingEmptyItems = CalculateMissingEmptyItems(elementsAddedInTheGird);
+            for (var i = 0; i < missingEmptyItems; i++)
+            {
+                var emptyItem = emptyItemsPool.Get();
+                emptyItem.gameObject.name = "EmptyItem";
+                instantiatedEmptyItems.Add(emptyItem);
+            }
+
+            LayoutRebuilder.ForceRebuildLayoutImmediate(view.EquippedItemsContainer);
+            LayoutRebuilder.ForceRebuildLayoutImmediate(scrollContainer);
+        }
+
+        private int CalculateMissingEmptyItems(int totalItems)
+        {
+            int remainder = totalItems % 6;
+            int missingItems = (remainder == 0) ? 0 : 6 - remainder;
+            return missingItems;
+        }
+
+        private async UniTaskVoid AwaitEquippedItemsPromiseAsync(WearablePromise equippedWearablesPromise, EmotePromise equippedEmotesPromise, CancellationToken ct)
+        {
+            var wearablesUniTaskAsync = await equippedWearablesPromise.ToUniTaskAsync(world, cancellationToken: ct);
+            var emotesUniTaskAsync = await equippedEmotesPromise.ToUniTaskAsync(world, cancellationToken: ct);
+
+            if (!wearablesUniTaskAsync.Result!.Value.Succeeded || !emotesUniTaskAsync.Result!.Value.Succeeded || ct.IsCancellationRequested)
+                return;
+
+            var currentWearables = wearablesUniTaskAsync.Result.Value.Asset.Wearables;
+            var currentEmotes = emotesUniTaskAsync.Result.Value.Asset.Emotes;
+            SetGridElements(currentWearables, currentEmotes);
+        }
+
+        private async UniTaskVoid WaitForThumbnailAsync(IWearable itemWearable, EquippedItem_PassportFieldView itemView, CancellationToken ct)
+        {
+            Sprite? sprite = await thumbnailProvider.GetAsync(itemWearable, ct);
```

Comment:
Nullable annotation is not enabled here: you can enable it but then you will need to handle everything properly in all other cases